### DISCUSSION TITLE
Multitask Prompt Tuning Confusion

### DISCUSSION THREAD

**Discussion by atastats** [huggingface/peft#1429] [Q&A]
Hey all,
Looking at the documentation on HF for multitask prompt tuning was not very helpful. I dug into the class definition for peft.MultitaskPromptTuningConfig and it seems like prompt_tuning_init_state_dict is how the task specific vanilla tuned source prompts are handled. However, I can't seem to figure out what the content of that dict should look like. Anyone have any insight into this or maybe some example of the using the class/method?

**Comment by BenjaminBossan** [comment]
You probably saw this [example notebook](https://github.com/huggingface/peft/blob/main/examples/conditional_generation/multitask_prompt_tuning.ipynb), but if not, please check it out.

Otherwise, I hope that @mayank31398 can give more insights.

**Reply by mayank31398** [reply]
@BenjaminBossan can you let me know the guidelines on adding documentation.
I think its long due and I want to start adding that.
Does adding docstrings to the code make it automatically appear on the [HF page](https://huggingface.co/docs/peft/en/package_reference/multitask_prompt_tuning)

**Reply by atastats** [reply]
This seems helpful. I hadn't seen this notebook... not sure why I was unable to find it. Looks like it has num_tasks set to 1, but I will try and extrapolate from there. Thank you!

**Reply by mayank31398** [reply]
Sure, I will try to document stuff soon.
This is a long standing stuff on my plate and unfortunately, I got looped into too much stuff.
Let me know if you run into any issues in this thread.

**Reply by BenjaminBossan** [reply]
Thanks @mayank31398. Yeah, adding a docstring to the classes listed in your link should make them appear automatically. Feel free to just create a PR, the docs should be built automatically and we can check there if they contain the newly added content.
